Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/scan/SubmissionQRCodeScanViewModel.kt

```diff
@@ -51,10 +52,11 @@ class SubmissionQRCodeScanViewModel @AssistedInject constructor(
     )
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal fun doDeviceRegistration(scanResult: QRScanResult) = launch {
+    internal fun doDeviceRegistration(coronaTestQRCode: CoronaTestQRCode) = launch {
         try {
             registrationState.postValue(RegistrationState(ApiRequestState.STARTED))
-            val testResult = submissionRepository.asyncRegisterDeviceViaGUID(scanResult.guid!!)
+            // TODO call new CoronaTestRepository
```

Comment:
Are you going to work on this TODO once the big PR is merged in this or in a subsequent PR?

Reply:
Yes, this won't compile anymore